pkg/yang/entry_test.go

```diff
@@ -338,6 +355,18 @@ func TestEntryNamespace(t *testing.T) {
 			ns:      "urn:baz",
 			wantMod: "baz",
 		},
+		{
+			descr:   "leaf directly defined within an augment to foo from submodule baz-augment of baz has baz's namespace",
+			entry:   foo.Dir["foo-c"].Dir["baz-submod-leaf"],
+			ns:      "urn:baz",
+			wantMod: "baz",
+		},
+		{
+			descr:        "leaf directly defined within an augment to foo from orphan submodule qux-augment has empty namespace",
+			entry:        foo.Dir["foo-c"].Dir["qux-submod-leaf"],
+			ns:           "",
+			wantModError: `could not find module "" when retrieving namespace for qux-submod-leaf`,
```

Comment:
This may or may not occur in reality, but here `Process` didn't return an error, so it looks fine to me.

I agree it seems strange to allow submodules to be parsed without supplying a module. Should `Process` have caught this error? If so, we should open an issue about it.